src/Illuminate/Foundation/ComposerScripts.php

```diff
@@ -41,6 +41,17 @@ protected static function clearCompiled()
     {
         $laravel = new Application(getcwd());
 
+        // Create environment file if it doesn't exists.
+        if (! file_exists($laravel->environmentFilePath())) {
+            exec("php -r \"copy('".$laravel->environmentFilePath().".example',
+             '".$laravel->environmentFilePath().');"');
```

Comment:
@GrahamCampbell the commands as listed [here](https://github.com/laravel/laravel/blob/master/composer.json#L32-L37) are only run when laravel installer is used to create the application. 

This behavior doesn't happen when i create a Laravel application through version control such as official [Laravel](https://github.com/laravel/laravel) repo, and run command `compose install`